### DISCUSSION TITLE
add `emplace_hint()` to API

### DISCUSSION THREAD

**Discussion by crusader-mike** [nlohmann/json#2437] [Ideas]
json has `emplace()` -- why not add `emplace_hint()`?

For example when you process data sorted by key and convert (a subset of) it into json -- in this case using `j.emplace_hint(j.end(), key, ...)` will allow me to avoid unnecessary lookups in underlying `std::map`.

**Comment by nlohmann** [comment]
Any idea what to do with arrays?

**Comment by crusader-mike** [comment]
throw a "bad object type" exception, I guess -- `emplace_hint()` assumes user expects associative container semantic
